src/content/en/fundamentals/progressive-web-apps/prpl.md

```diff
@@ -0,0 +1,243 @@
+project_path: /web/_project.yaml
+book_path: /web/fundamentals/_book.yaml
+description: 
+
+{# wf_updated_on: 2016-09-28 #}
+{# wf_published_on: 2016-09-28 #}
+
+# The PRPL Pattern {: .page-title }
+
+{% include "web/_shared/contributors/addyosmani.html" %}
+
+Dogfood: PRPL is a pattern we believe makes sense for applications that can be
+granularly served. It is currently considered experimental, however we plan on
+publishing more data on where it works well in the future.
+
+The mobile web is too slow. Over the years the web has evolved from a
+document-centric platform to a first-class application platform. Thanks to
+advancements in the platform itself (such as [Service
+Workers](/web/fundamentals/primers/service-worker/)) and in the tools and
+techniques we use to build apps, users can do virtually anything on the web they
+can do in a native app.
+
+But at the same time, the bulk of our computing has moved from powerful desktop
+machines with fast, reliable network connections to relatively underpowered
+mobile devices with connections that are often slow, flaky or both. This is
+especially true in parts of the world where the next billion users are coming
+online.
+
+Unfortunately, the patterns we devised for building and deploying powerful,
+feature-rich web apps in the desktop era generally yield apps that take far too
+long to load on mobile devices – so long that many users simply give up.
+
+This presents an opportunity to craft new patterns that take advantage of modern
+web platform features to granularly deliver mobile web experiences more quickly.
+PRPL is one such pattern.
+
+## The PRPL pattern
+
+
+PRPL is a pattern for structuring and serving Progressive Web Apps (PWAs), with
+an emphasis on the performance of app delivery and launch. It stands for:
+
+*  **Push** critical resources for the initial route.
+*  **Render** initial route.
+*  **Pre-cache** remaining routes.
+*  **Lazy-load** and create remaining routes on demand.
```

Comment:
I'm not sure I understand how you're using routes in this list? It seems like you might mean resources, but then the first sentence wouldn't make sense.